docs/rules/no-implicit-service-injection-argument.md

```diff
@@ -0,0 +1,40 @@
+# no-implicit-service-injection-argument
+
+:wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+Disallow omitting the service name argument when injecting a service.
+
+Some developers may prefer to be more explicit about what service is being injected instead of relying on the implicit and potentially costly lookup/normalization of the service from the property name.
+
+## Examples
+
+Examples of **incorrect** code for this rule:
+
+```js
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+
+export default class Page extends Component {
+  @service() serviceName;
+}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+
+export default class Page extends Component {
+  @service('service-name') serviceName;
```

Comment:
@rwjblue both `@service('service-name')` or `@service('serviceName')` are acceptable right?

Reply:
More discussion in: https://discord.com/channels/480462759797063690/485447409296736276/842127236198039593

Reply:
I've updated the rule to autofix to the kebab-case service name (i.e. `@service('service-name')`) since kebab-case filenames are more common.  In the future, once we can actually resolve services from inside the lint rule, we can make this use the actual resolved service name.